SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/Storage/Storage.cs

```diff
@@ -224,7 +291,19 @@ await _storageAdapter.GetAsync<LogEntry>(entry => entry.Channel == channelName,
                 ProcessLogIds(channelName, batchId, idPairs);
                 logs?.AddRange(retrievedLogs);
                 return batchId;
+            });
+
+            try
+            {
+                _queue.Add(task);
             }
+            catch (InvalidOperationException)
+            {
+                throw new StorageException("The operation has been cancelled");
+            }
+
+            await FlushQueue().ConfigureAwait(false);
```

Comment:
It would work better to just have a unique FlushQueue call that continously run in background and never return and remove all `await FlushQueue().ConfigureAwait(false);` calls to avoid that. Plus all of your `FlushQueue()` are competing with each other, this is overkill.